docs/spec/auth/scope.md

```diff
@@ -56,7 +56,7 @@ it.
 The resource name represent the name which identifies a resource for a resource
 provider. A resource is identified by this name and the provided resource type.
 An example of a resource name would be the name component of an image tag, such
-as "samalba/myapp".
+as "samalba/myapp" or "hostname/samalba/myapp".
```

Comment:
Could `samalba` be interpreted as a hostname in the first example? Would the ambiguity cause any problems?

I think the current `reference` package treats the first component as a hostname if it validates as a DNS name, but Docker Engine has a different approach for parsing hostnames, and only considers the first component a hostname if it has a port number or a `.` character.

It's kind of confusing :(